core/src/main/java/org/apache/gravitino/connector/BaseCatalog.java

```diff
@@ -181,53 +182,64 @@ public CatalogOperations ops() {
+      return;
+    }
+
     String authorizationProvider =
         catalogPropertiesMetadata().containsProperty(AUTHORIZATION_PROVIDER)
             ? (String) catalogPropertiesMetadata().getOrDefault(conf, AUTHORIZATION_PROVIDER)
             : null;
 
     if (authorizationProvider == null) {
       LOG.info("Authorization provider is not set!");
-      return null;
+      return;
     }
 
-    ServiceLoader<AuthorizationProvider> loader =
-        ServiceLoader.load(
-            AuthorizationProvider.class, Thread.currentThread().getContextClassLoader());
-
-    List<Class<? extends AuthorizationProvider>> providers =
-        Streams.stream(loader.iterator())
-            .filter(p -> p.shortName().equalsIgnoreCase(authorizationProvider))
-            .map(AuthorizationProvider::getClass)
-            .collect(Collectors.toList());
-    if (providers.isEmpty()) {
-      throw new IllegalArgumentException(
-          "No authorization provider found for: " + authorizationProvider);
-    } else if (providers.size() > 1) {
-      throw new IllegalArgumentException(
-          "Multiple authorization providers found for: " + authorizationProvider);
-    }
+    BaseAuthorization<?> baseAuthorization;
     try {
-      return (BaseAuthorization<?>)
-          Iterables.getOnlyElement(providers).getDeclaredConstructor().newInstance();
+      baseAuthorization =
+          classLoader.withClassLoader(
+              cl -> {
+                try {
+                  ServiceLoader<AuthorizationProvider> loader =
+                      ServiceLoader.load(AuthorizationProvider.class, cl);
+
+                  List<Class<? extends AuthorizationProvider>> providers =
+                      Streams.stream(loader.iterator())
+                          .filter(p -> p.shortName().equalsIgnoreCase(authorizationProvider))
+                          .map(AuthorizationProvider::getClass)
+                          .collect(Collectors.toList());
+                  if (providers.isEmpty()) {
+                    throw new IllegalArgumentException(
+                        "No authorization provider found for: " + authorizationProvider);
+                  } else if (providers.size() > 1) {
+                    throw new IllegalArgumentException(
+                        "Multiple authorization providers found for: " + authorizationProvider);
+                  }
+                  return (BaseAuthorization<?>)
+                      Iterables.getOnlyElement(providers).getDeclaredConstructor().newInstance();
+                } catch (Exception e) {
+                  LOG.error("Failed to create authorization instance", e);
+                  throw new RuntimeException(e);
+                }
+              });
     } catch (Exception e) {
-      LOG.error("Failed to create authorization instance", e);
+      LOG.error("Failed to load authorization with class loader", e);
       throw new RuntimeException(e);
     }
+
+    if (baseAuthorization == null) {
```

Comment:
so it seems that if the code goes here, `baseAuthorization` is not null.

Reply:
Removed these codes.